server/controllers/index.js

```diff
@@ -2,9 +2,11 @@ const express = require('express')
 const router = express.Router()
 const { getAllEvents } = require('./getAllEvents')
 const userEvents = require('./userHome')
+const addEvent = require('../controllers/addEvent')
 
 router.get('/api/user-events/:id', userEvents)
 
 router.get('/api/events', getAllEvents)
+router.post('/api/booking', addEvent)
```

Comment:
'/api/events' is to get all events we have..
This is new for me too, she should add id parameter to '/api/events', so should it fall under the '/api/events' too?

Reply:
Right, good point, I missed the `user-events` API. Yes it should also fall under the `events` API.
Here is a quick overview of RESTful API design:
- it is organized around resources (something that the application manages), typically a resource corresponds to one table in the database (does not mean that every database table is a resource!). So, in your case, 'events' are one type of resource, 'users' another one and 'rooms' the last one
- the API design then revolves around these resources and has consistent semantics. The base URL for all of the APIs for a certain resource consists of the service URL (`/api` in your case) and the resource collection (basically the resource name in plural, `/events`). APIs that operate on a specific resource instead of on the whole collection then append the ID (`/{resource_id}`).
- here are the standard APIs for any resource (with the HTTP method to use):
  - CREATE (`POST /{service_URL}/{resource}`)
  - LIST (`GET /{service_URL}/{resource}`)
  - GET (`GET /{service_URL}/{resource}/{resource_id}`)
  - UPDATE (`PATCH /{service_URL}/{resource}/{resource_id}`). People often use POST instead of PATCH
  - DELETE (`DELETE /{service_URL}/{resource}/{resource_id}`)
- you might not need all of them (for example you might not want to allow the clients to update or delete the resource once created so you don't provide those methods). Notice that the `GET` API operates on a concrete resource, ie. you provide an ID (`GET /api/events/123`) and the API will return data for that specific resource. `LIST` on the other hand operates on the collection level and is basically a search. If you don't provide any query parameters, it will return all of the existing resources (this should be limited to a certain MAX number but that's another topic). Thus, if you want to find all the events belonging to a certain user, you should call it like this: `GET /api/events?userId=123`. The `userId` is a query parameter that you will implement in the API.

That's it in a nutshell, obviously simplified, hope it makes sense. There are additional things to consider and take care of, but I think this should be good to start with. Here is a [guide ](https://cloud.google.com/apis/design) that Google uses for designing their APIs, there should be plenty of other sources as well. Don't hesitate to ask questions!

Reply:
just to follow up with an example, here is the flow to create an event:

1. first you have to create the user and the room resources - if they haven't been created yet. Thus you would call the appropriate Create APIs on the `user` and `room` resources. The create APIs will return the IDs of the created resources. Alternatively, if these resources already exist, you would call the List APIs with appropriate search parameters to get their IDs.
2. You would then use these IDs in the call to create the event resource
```
POST /api/events

{ event_name: 'Test Event', user_id: ${userID}, room_id: ${roomID} ..... }
```